source/extensions/upstreams/tcp/generic/config.cc

```diff
@@ -16,7 +17,11 @@ TcpProxy::GenericConnPoolPtr GenericConnPoolFactory::createGenericConnPool(
     TcpProxy::TunnelingConfigHelperOptConstRef config, Upstream::LoadBalancerContext* context,
     Envoy::Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks,
     StreamInfo::StreamInfo& downstream_info) const {
-  if (config.has_value()) {
+  const Router::StringAccessor* disable_tunneling =
```

Comment:
This block should be moved under `config.has_value()` to avoid lookup for non-tunnel TCP streams.

Reply:
done